libguard/guard_interface.cpp

```diff
@@ -178,17 +193,32 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
 
     // Space left in GUARD file before writing a new record
     avalSize = file.size() - ((lastPos + 1) * sizeof(existGuard) + headerSize);
-	if (avalSize < sizeof(existGuard))
+    if (avalSize < sizeof(existGuard))
     {
-        guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", avalSize);
-        throw std::runtime_error("Enough size is not available in GUARD file");
+        if ((empPos == 0) && !isResolvedRecordPresent)
+        {
+            guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", avalSize);
+            throw std::runtime_error(
+                "Enough size is not available in GUARD file");
+        }
+
+        resolveRecordClear = true;
     }
 
     GuardRecord guard;
     //! if blank record exist
     if (isBlankRecord(existGuard))
     {
-        offset = lastPos * sizeof(guard);
+        // If oldest resolved record is deleted, than new record will be
+        // written in that empty slot
+        if (resolveRecordClear)
```

Comment:
Cannot we use the LN 196 (new code) block to set `offset`?

Reply:
I prefer adding it here rather in LN 196 to avoid confusion.

Reply:
OK, I would prefer in LN196 so that we can avoid one more `if and else` block and also, IMO when we are doing something different based on the condition then that can go into the same block statement (if possible) to give more readability.  

Any advantage to prefer this way?